### DISCUSSION TITLE
How to manage conflicts in Git Extensions ?

### DISCUSSION THREAD

**Discussion by Whaouu** [gitextensions/gitextensions#11607] [Q&A] [answered]
Hello,

At the moment, I am testing "Git Extensions" with the aim of doing a POC. The purpose of the demonstration is to:

- Create a working branch that is based on the master.
- Work on the new branch. Make commits/pushes there.
- Update the working branch with the latest changes from the "master" branch in order to resolve any conflicts before pushing the files from the working branch to the master.
- After resolving conflicts, push the changes from the working branch to the master.
- Delete the working branch.

I have done some tests on the command line. It may not be very optimized, but I get the desired result:

Creating a branch based on master and positioning on the new branch:
```
git checkout master
git branch 23_02_2024
git checkout 23_02_2024
```

The user updates the files on the new branch. He sends them to the GIT followed by a commit / push in the branch "23_02_2024":
```
git add .
git commit -m "BR17"
git push origin 23_02_2024
```

Updating the local branch "23_02_2024" with the latest changes from the "master" branch:
```
git fetch origin master
git merge origin/master
```

Conflict resolution:
```
git status
# Modify the file(s) to resolve the conflict
git add .
git commit -m "Conflict resolution"
```

Updating the master branch and synchronization with "23_02_2024":
```
git checkout master
git pull
git checkout 23_02_2024
git fetch origin master
git merge origin/master
git push origin 23_02_2024
```

Merging the local branch to master:
```
git checkout master
git merge 23_02_2024
git push origin master
```

I would like to redo these operations via the Git Extensions interface and there, I get lost in the menus, options, etc.

I would need a good connoisseur of Git Extension who can guide me step by step in the steps to get the same result as on the command line.

Thank you in advance for your help.

**Comment by gerhardol** [comment]
Described some in the documentation
https://git-extensions-documentation.readthedocs.io/en/release-4.2/modify_history.html#merge-conflicts

**Reply by Whaouu** [reply]
Hello,

Thank you for your response. Looking at the documentation was the first thing I did, but I didn’t find enough explanations.

I allow myself to put here the steps (with the corresponding screenshots) that I performed in Git Extensions:

1 - Updating the working branch with the latest modifications from the “master” branch in order to resolve any conflicts before pushing the files from the working branch to the master.

![1](https://github.com/gitextensions/gitextensions/assets/17141985/306652ad-bd9a-4166-99fc-8e6647482934)

2 - On the conflict resolution window, I press CTRL+2 to choose to keep the REMOTE file.

![2](https://github.com/gitextensions/gitextensions/assets/17141985/ca68267d-b76d-4c2a-9114-edec2f35cc21)

3 - A window named “Patch apply” appears. I click on the YES button.

![3](https://github.com/gitextensions/gitextensions/assets/17141985/4f259536-0825-40e3-84fe-68998030fb71)

4 - Another window appears. I don’t know what I should do.

![4](https://github.com/gitextensions/gitextensions/assets/17141985/b457bae1-ad5e-4742-95c2-3dfab7e2e1b2)

Can you help me?

**Reply by pmiossec** [reply, marked as answer]
In more than 10 years using and contributing to GitExtensions, I never saw this "You are in the middle of a patch apply, continue patch apply?" popup and suspect that your repository is in a bad state.

Did you tried to apply a patch through GitExtensions or git command line and never finished applying it?

I would say, click on "Abort patch" in the last window and try again doing your "pull merge".

The other possibility is that it is normal because git fail to apply itself the patch when you do "CTRL+2" (that I don't use often, so that could explain why I never encountered this case) due to conflicts. If it's the case, I think that clicking on "Solve conflicts" is the way to go...

**Comment by Whaouu** [comment]
Hello Pmiossec,

Sorry for my very late return and thank you very much for your reply.

There was indeed a problem on this repository. I simply deleted the folder on my workstation and re-synchronized to restore normal behavior.
